examples/datasource-http-backend/pkg/plugin/datasource.go

```diff
@@ -39,6 +42,12 @@ func NewDatasource(settings backend.DataSourceInstanceSettings) (instancemgmt.In
 	if err != nil {
 		return nil, fmt.Errorf("http client options: %w", err)
 	}
+	// http clients created with httpclient.New have a middleware stack consisting
+	// of useful generic middlewares such as:
+	//  - TracingMiddleware (spans creates for each outgoing HTTP request)
```

Comment:
nit: indentation here (and maybe move the TracingMiddleware to the last, since it's the most recent)

Reply:
I think those highlighted by @marefr are important to have documented somewhere, as it took me a while to understand what they did the first time I saw them, without reading the source code. We can add it to the example but maybe we should also have it in the docs somewhere?

Maybe we can have a new page under "Backend plugins" explaining how to use HTTP clients on backend plugins? However I believe I already saw something similar a while ago (maybe one of the plugin authentication docs), but I'm not sure.

WDYT?

Reply:
the more documentation, the better :shrug:

Reply:
I'd say we can add what Marcus suggested to this plugin example for now to have it somewhere and then see if it's worth having it in the docs as well